### DISCUSSION TITLE
[Help] Theme listview not working anymore after upgrade to 1.6.1 (from 1.5.4)

### DISCUSSION THREAD

**Discussion by cpiber** [davatorium/rofi#1365] [Q&A] [answered]
I have the following theme:

```
configuration {
    show-icons: true;
    font: "Hack 10";
}

@import "colors.rasi"

* {
    /*selected-background: #d8c45d;
    selected-foreground: black;*/
    border-color: #e60053;
}

#window {
    width: 550px;
    height: 80px;
    background-color: @background-solid;
    padding: 12px;
    border-radius: 6px;
}

#mainbox {
    border:  0;
    padding: 0;
    children: [listview];
    background-color: @background-solid;
}

#listview {
    columns: 4;
    layout: vertical;
    background-color: transparent;
}

element {
    margin: 8px;
    padding: 10px 4px;
    orientation: vertical;
    background-color: @background-solid;
    text-color:       @foreground;
}

element-icon {
    size: 48px;
    margin: 2px;
}

element-text {
    horizontal-align: 0.5;
    vertical-align: 0.5;
    text-align: center;
}

#element.normal.normal {
}

#element.selected.normal {
    border: 0 0 2px;
}

#element.alternate.normal {
}

#mode-switcher {
    border:       0px 0px 0px ;
    border-color: @separatorcolor;
}

#button {
    spacing:    0;
    text-color: @foreground;
}

#button.selected {
    background-color: @selected-background;
    text-color:       @selected-foreground;
}
```

On 1.5.4, this would produce a vertical list as expected. On 1.6.1, I only get an empty rectangle, as if the items weren't there.

I'm not familiar with rofi theming, I got this theme from reddit somewhere, so any help is appreciated.

Also, in the documentation, it is mentioned that all paths should start with `#`, but nowhere do they actually use them, and this theme seems to mix them, what is the correct way?

Edit: It seems now that `show-icons: true` is the problem, how could I fix this in 1.6.1? Without this I cannot seem to get icons to appear

**Comment by DaveDavenport** [comment, marked as answer]
can you paste the rest of the theme? so I can test.

**Reply by cpiber** [reply]
That is everything in this theme.

colors.rasi:
```
* {
    foreground:           #dfdfdf;
    selected-foreground:  #555;

    background:           #222; 
    background-solid:     #222; 
    selected-background:  #444;

    red:                  rgba ( 220, 50, 47, 100 % );
    blue:                 rgba ( 38, 139, 210, 100 % );
}

```

Content is generated with:
```
    echo -en "Shutdown\0icon\x1fsystem-shutdown\n"
    echo -en "Logout\0icon\x1fsystem-log-out\n"
    echo -en "Suspend\0icon\x1fsystem-suspend\n"
    echo -en "Reboot\0icon\x1fsystem-restart\n"
```

**Reply by DaveDavenport** [reply]
thanks.

You don't see any elements because you force the window to be to small for its content.

**Reply by cpiber** [reply]
As I said, it worked before, in 1.5.4, why doesn't it work anymore? I specifically gave the icons a size, and even with `size: 0px` nothing shows up

**Reply by DaveDavenport** [reply]
It works here if there is enough size to fit the icon. e.g. making window higher, or icon smaller. 
1.5.4 had several bugs in drawing that might make a broken theme, kinda work. Also icon size in listview worked differently in 1.5.4, so there is not a 1 on 1 comparison.

(`0px` indeed does not work, that might be a bug (but why would you set size to 0???, 1px works for me in git)

![rofi-2021-08-08-1433-00000](https://user-images.githubusercontent.com/406245/128632109-c7d65d06-b777-4993-9185-367d2d20908a.png)

**Reply by cpiber** [reply]
I wanted to set it to `0px` to test. Setting `element-icon`'s `size` to `1px` does not help either. But removing the width and height constraints, I found that the icon is above, when it should be next to the text...

Thank you for your time!
